src/ensembl/production/metadata/service.py

```diff
@@ -458,6 +458,45 @@ def release_by_uuid_iterator(metadata_db, genome_uuid):
         yield create_release(dict(result))
 
 
+def get_dataset_by_genome_id(metadata_db, genome_uuid, requested_dataset_type):
+    # This is sqlalchemy's metadata, not Ensembl's!
+    md = db.MetaData()
+    session = Session(metadata_db, future=True)
+
+    # Reflect existing tables, letting sqlalchemy load linked tables where possible.
+    genome = db.Table('genome', md, autoload_with=metadata_db)
+    genome_dataset = db.Table('genome_dataset', md, autoload_with=metadata_db)
+    dataset = db.Table('dataset', md, autoload_with=metadata_db)
+    dataset_type = db.Table('dataset_type', md, autoload_with=metadata_db)
+    attribute = db.Table('attribute', md, autoload_with=metadata_db)
+    dataset_attribute = db.Table('dataset_attribute', md, autoload_with=metadata_db)
+    ensembl_release = db.Table('ensembl_release', md, autoload_with=metadata_db)
+
+    dataset_select = db.select(
+                               dataset.c.dataset_uuid,
+                               dataset.c.name.label('dataset_name'),
+                               attribute.c.name,
+                               dataset_attribute.c.type,
+                               dataset.c.version.label('dataset_version'),
+                               dataset.c.label.label('dataset_label'),
+                               ensembl_release.c.version,
+                               dataset_attribute.c.value
+                               ) \
+        .select_from(genome).filter_by(genome_uuid=genome_uuid) \
+        .join(genome_dataset) \
+        .join(ensembl_release) \
+        .join(dataset) \
+        .join(dataset_attribute) \
+        .join(attribute) \
+        .join(dataset_type) \
+        .where(dataset_type.c.name == requested_dataset_type) \
+        .order_by(dataset_type.c.name, dataset.c.name) \
```

Comment:
Is the order by necessary here?

Reply:
Not sure for now. Need further discussion.

Reply:
If that's the case, I think there is no harm in keeping it for now